### PR TITLE
[PULL REQUEST] Add RO2concAfterChem diagnostic

### PR DESCRIPTION
**Pull request by msulprizio**
This PR closes https://github.com/geoschem/geos-chem/issues/1265 by enabling the RO2concAfterChem diagnostic for all full-chemistry simulations. Previously, this was available only when `MODEL_GEOS` was used. The calculation of RO2 has been removed from fullchem_mod.F90 and instead we define RO2 as a family in fullchem.kpp.

### CODE REVIEW

File: KPP/fullchem/fullchem.kpp [hunk 1]
```diff
@@ -17,6 +17,7 @@ LCO : CO;
 PSO4 : SO4;
 LCH4 : CH4;
 PH2O2 : H2O2;
+PRO2: A3O2 + ATO2 + B3O2 + BRO2 + ETO2 + HO2 + LIMO2 + MO2 + PIO2 + PO2 + PRN1 + R4N1 + R4O2 + TRO2 + XRO2 + IHOO1 + IHOO4 + ICHOO + IHPOO1 + IHPOO2 + IHPOO3 + IEPOXAOO + IEPOXBOO + C4HVP1 + C4HVP2 + HPALD1OO + HPALD2OO + ISOPNOO1 + ISOPNOO2 + INO2D + INO2B + IDHNBOO + IDHNDOO1 + IDHNDOO2 + IHPNBOO + IHPNDOO + ICNOO + IDNOO;
```
Comment (yantosca) [inline KPP/fullchem/fullchem.kpp:20]: Looks good.  Will also have to add this to the `KPP/custom/custom.kpp` fille as well.

**Review by yantosca — APPROVED** [review]
This all looks OK, as it is adding a new diagnostic via KPP.  This greatly simplifies the code as well.

**Comment by yantosca** [comment]
Made a few updates:
  1. Merged the `dev/14.1.0` branch into `feature/ro2-diag` to bring these modifications up to the 14.1.0 development branch
  2. Added an entry for PRO2 into `species_database.yml`

**Comment by yantosca** [comment]
After creating new restart files with an additional variable `SpeciesRst_PRO2` (set to zero everywhere), a rerun of the GEOS-Chem Classic integration tests passed.

```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

Using 24 OpenMP threads
Number of execution tests: 48
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gc_025x03125_CH4_geosfp_47L_na...................Execute Simulation.....PASS
gc_025x03125_fullchem_geosfp_47L_na..............Execute Simulation.....PASS
gc_05x0625_CH4_merra2_47L_na.....................Execute Simulation.....PASS
gc_05x0625_fullchem_merra2_47L_na................Execute Simulation.....PASS
gc_2x25_aerosol_merra2...........................Execute Simulation.....PASS
gc_2x25_CH4_merra2...............................Execute Simulation.....PASS
gc_2x25_CO2_merra2...............................Execute Simulation.....PASS
gc_2x25_fullchem_merra2..........................Execute Simulation.....PASS
gc_2x25_Hg_merra2................................Execute Simulation.....PASS
gc_2x25_metals_merra2............................Execute Simulation.....PASS
gc_2x25_POPs_BaP_merra2..........................Execute Simulation.....PASS
gc_2x25_tagCH4_merra2............................Execute Simulation.....PASS
gc_2x25_tagCO_merra2.............................Execute Simulation.....PASS
gc_2x25_tagO3_merra2.............................Execute Simulation.....PASS
gc_2x25_TransportTracers_merra2..................Execute Simulation.....PASS
gc_2x25_TransportTracers_merra2_LuoWd............Execute Simulation.....PASS
gc_4x5_aerosol_geosfp............................Execute Simulation.....PASS
gc_4x5_aerosol_merra2............................Execute Simulation.....PASS
gc_4x5_CH4_geosfp................................Execute Simulation.....PASS
gc_4x5_CH4_merra2................................Execute Simulation.....PASS
gc_4x5_fullchem_aciduptake_merra2................Execute Simulation.....PASS
gc_4x5_fullchem_APM_merra2.......................Execute Simulation.....PASS
gc_4x5_fullchem_benchmark_merra2.................Execute Simulation.....PASS
gc_4x5_fullchem_complexSOA_merra2................Execute Simulation.....PASS
gc_4x5_fullchem_complexSOA_SVPOA_merra2..........Execute Simulation.....PASS
gc_4x5_fullchem_geosfp...........................Execute Simulation.....PASS
gc_4x5_fullchem_marinePOA_merra2.................Execute Simulation.....PASS
gc_4x5_fullchem_merra2...........................Execute Simulation.....PASS
gc_4x5_fullchem_merra2_47L.......................Execute Simulation.....PASS
gc_4x5_fullchem_merra2_LuoWd.....................Execute Simulation.....PASS
gc_4x5_fullchem_RRTMG_merra2.....................Execute Simulation.....PASS
gc_4x5_fullchem_TOMAS15_merra2_47L...............Execute Simulation.....PASS
gc_4x5_fullchem_TOMAS40_merra2_47L...............Execute Simulation.....PASS
gc_4x5_Hg_geosfp.................................Execute Simulation.....PASS
gc_4x5_Hg_merra2.................................Execute Simulation.....PASS
gc_4x5_metals_merra2.............................Execute Simulation.....PASS
gc_4x5_POPs_BaP_geosfp...........................Execute Simulation.....PASS
gc_4x5_POPs_BaP_merra2...........................Execute Simulation.....PASS
gc_4x5_tagCH4_geosfp.............................Execute Simulation.....PASS
gc_4x5_tagCH4_merra2.............................Execute Simulation.....PASS
gc_4x5_tagCO_geosfp..............................Execute Simulation.....PASS
gc_4x5_tagCO_merra2..............................Execute Simulation.....PASS
gc_4x5_tagO3_geosfp..............................Execute Simulation.....PASS
gc_4x5_tagO3_merra2..............................Execute Simulation.....PASS
gc_4x5_TransportTracers_geosfp...................Execute Simulation.....PASS
gc_4x5_TransportTracers_geosfp_LuoWd.............Execute Simulation.....PASS
gc_4x5_TransportTracers_merra2...................Execute Simulation.....PASS
gc_4x5_TransportTracers_merra2_LuoWd.............Execute Simulation.....PASS
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 48
Execution tests failed: 0
Execution tests not yet completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Comment by yantosca** [comment]
After creating new restart files with the `SPC_PRO2` variable set to zero everywhere, a rerun of the GCHP integration tests all passed:
```console
==============================================================================
GCHP: Execution Test Results

Number of execution tests: 3
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_fullchem_benchmark_merra2_c48...............Execute Simulation.....PASS
gchp_fullchem_standard_merra2_c24................Execute Simulation.....PASS
gchp_TransportTracers_geosfp_c24.................Execute Simulation.....PASS
 
Summary of execution test results:
------------------------------------------------------------------------------
Execution tests passed:        3
Execution tests failed:        0
Execution tests not completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
NOTE: New restart files for GCClassic and GCHP fullchem and TOMAS Simulations will be stored in the path `ExtData/GEOSCHEM_RESTARTS/v2022-10`.   Carboncycle restart files will also go into this folder.

**Comment by msulprizio** [comment]
Note: This update causes slight differences in the full-chemistry benchmark simulations at the level of numerical noise attributed to adding a new prod-loss species into the KPP mechanism. We will put off merging in the PR for now until we decide (1) if these changes are acceptable and (2) if this diagnostic is widely needed by the community.

**Comment by msulprizio** [comment]
I will close out this pull request for now. If it is deemed a priority by GEOS-Chem users/working groups then we can reopen it at a later date.

**Pull request closed without merging**